### DISCUSSION TITLE
Add new predicate for required column for entry Namespaces in StructTreeRoot

### DISCUSSION THREAD

**Discussion by MaximPlusov** [pdf-association/arlington-pdf-model#114] [Ideas]
ISO 32000-2:2020, Table 354: `Namespaces - Required if any structure elements have namespace identifiers.`

Maybe it is possible to add new predicate to check this requirment?

There may be many such cases not represented in the Arlington. Maybe they need to be documented somehow?

**Comment by petervwyatt** [comment]
I think we'd have to invert that requirement by adding a "SpecialCase" predicate to **NS** in `StructureElement` that asserts something like `fn:IsPresent(trailer::Catalog::StructTreeRoot::Namespaces)`.

AFAICT there also isn't any strict "shall" requirement in ISO 32000-2 which says the **NS** in `StructureElement` has to be an element in the `StructTreeRoot` **Namespaces** array! At most, there is a descriptive semi-factual statement in Table 354: "_An array of namespaces used within the document._". This possibly seems like an oversight...

**Reply by petervwyatt** [reply]
See also [PDF Errata #396](https://github.com/pdf-association/pdf-issues/issues/396)
